internal/cloudsql/refresh.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2020 Google LLC
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     https://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cloudsql
+
+import (
+	"context"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+
+	sqladmin "google.golang.org/api/sqladmin/v1beta4"
+)
+
+// metadata contains information about a Cloud SQL instance needed to create connections.
+type metadata struct {
+	ipAddrs      map[string]string
+	serverCaCert *x509.Certificate
+	version      string
+}
+
+// fetchMetadata uses the Cloud SQL Admin API's get method to retreive the information about a Cloud SQL instance
+// that is used to create secure connections.
+func fetchMetadata(ctx context.Context, client *sqladmin.Service, inst connName) (metadata, error) {
```

Comment:
So the goal with these functions was to keep things functional and avoid mutating anything. IMO, it's less clear that's the case if you tie these functions to a struct (it's also easier to test like this so I can inject the dependency as needed). 

For example, we could potentially mock the sqladmin client here to avoid running these tests on the network.

Reply:
I can go either way on this. I have a slight preference to passing in an interface instead of a `*sqladmin.Service` just because mocking that type will be difficult (e.g., considering what it would take to mock a type whose use is `client.Instances.Get(inst.project, inst.name).Context(ctx).Do()`).

Reply:
Since this is a private method, we can easily change this signature in the future if testing with the sqladmin.Service becomes onerous.